keps/sig-autoscaling/20190307-configurable-scale-velocity-for-hpa.md

```diff
@@ -85,29 +88,30 @@ As a result, users can't influence scale velocity, and that is a problem for man
 
 ### Non-Goals
 
-TBA
+- Persist the scaling events so that the HPA behavior is consistent even when the controller is restarted.
 
 ## Proposal
 
-We need to introduce an algorithm-agnostic HPA object configuration that will configure each particular HPA scaling behavior.
-For both direction (scale up and scale down) there should be a `Constraint` field with the following parameters:
+We need to introduce an algorithm-agnostic HPA object configuration that will allow configuration of individual HPAs.
+To customize the scaling behavior we should add a `behavior` object with the following fields:
 
-- `periodSeconds` - a parameter to specify the time period for the constraint, in seconds
-- `percent` - a parameter to specify the relative speed, in percentages:
-    i.e., if ScaleUpPercent = 150 , then we can add 150% more pods (10 -> 25 pods)
-- `pods` - a parameter to specify the absolute speed, in the number of pods:
-    i.e., if ScaleUpPods = 5 , then we can add 5 more pods (10 -> 15)
-- `delay` - a parameter to specify the window over which the max (or min) recommendation is used. It behaves the same as the [Stabilization Window][].
+- `stabilizationWindowSeconds` - this value indicates the amount of time the HPA controller should consider
+  previous recommendations to prevent flapping of the number of replicas.
+- `scaleUp` specifies the rules which are used to control scaling behavior while scaling up.
+  - `choosePolicy` can be `min` or `max` and specifies which value from the policies should be selected. The `max` value is used by default.
+  - `policies` a list of policies which regulate the amount of scaling. Each item has the following fields
+    - `type` can have the value `pods` or `percent` which indicates the allowed changed in terms of absolute number of pods or percentage of current replicas.
```

Comment:
if we want to generalize this, it should be not "pods vs percent", but "absolute vs relative". Though, I won't fight for this change.

Reply:
Using `absolute`/`relative` implies there are only 2 choices where as more could be added in the future. So I would keep it as it is for now.